tools/coqdoc/output.ml

```diff
@@ -260,6 +260,40 @@ module Latex = struct
   let label_ident s =
     for i = 0 to String.length s - 1 do label_char s.[i] done
 
+
+  let convert_string f s =
+    let n = String.length s in
+    let rec qmap i =
+      if i == n then "" else (f s.[i]) ^ (qmap (i + 1))
+    in qmap 0
+
+  let fragment_char c =
+      match c with
+        | 'a'..'z' | 'A'..'Z' | '0'..'9' | '.' | '_'
+             | '<' | '>' | '&' | '\'' | '"' | '-' | ':' -> true
+        | _ -> false
+
+   let not_fragment_char c = not (fragment_char c)
+
+   let string_contains f s =
+     let n = String.length s in
+     let rec find i =
+        if i == n then false else if f s.[i] then true else find (i+1)
+      in find 0
+
+  let encode_fragment s =
+     if string_contains (not_fragment_char) s
+     then Digest.to_hex (Digest.string s)
+     else
+       convert_string (fun c ->
+         if c >= '\x80' then Printf.sprintf "x%X" (int_of_char c) else
```

Comment:
Unless I am missing something, this case should already have been handled by the `Digest.string` case.